src/frontend/hooks/useSecureStorage.ts

```diff
@@ -0,0 +1,39 @@
+import {setItemAsync, getItemAsync} from 'expo-secure-store';
+import {useState, useEffect, useCallback} from 'react';
+
+type SecureStoreKey = 'RootKey';
+
+export function useSecureStore(key: SecureStoreKey) {
+  const [value, setValue] = useState<string | null>(null);
+  const [status, setStatus] = useState<'idle' | 'loading'>('loading');
+
+  useEffect(() => {
+    getItemAsync(key)
+      .then(storageVal => {
+        setValue(storageVal);
+      })
+      .finally(() => {
+        setStatus('idle');
+      });
+  }, [key, getItemAsync]);
```

Comment:
probably doesn't affect anything but kind of confusing to have `getItemAsync` here since it's a purely stable reference as an imported function (i.e. not defined within the component)

```suggestion
  }, [key]);
```

Reply:
addressed https://github.com/digidem/CoMapeo-mobile/pull/89/commits/e8deba6c5f0d8705f606d5e1e2a158e2cdc706d2